# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSIOLayer.cpp
# Commit: 18651e373837
# Full Hash: 18651e373837007e879e22c9573f266698752a9a
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-22 15:53:06
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Make sure that nsHttpConnection immediately knows that the handshake is done. r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123919
# ==============================================================================

diff -r f8d31e3ebc93 -r 18651e373837 security/manager/ssl/nsNSSIOLayer.cpp
--- a/security/manager/ssl/nsNSSIOLayer.cpp	Wed Sep 22 09:19:31 2021 +0000
+++ b/security/manager/ssl/nsNSSIOLayer.cpp	Wed Sep 22 09:19:31 2021 +0000
@@ -251,17 +251,10 @@
 
   mIsFullHandshake = false;  // reset for next handshake on this connection
 
-  // HandshakeDone needs to be dispatched so that it is not called inside
-  // nss locks.
-  RefPtr<nsNSSSocketInfo> self(this);
-  NS_DispatchToCurrentThread(NS_NewRunnableFunction(
-      "nsNSSSocketInfo::HandshakeDone",
-      [self{std::move(self)}]() {
-        if (self->mTlsHandshakeCallback) {
-          auto callback = std::move(self->mTlsHandshakeCallback);
-          Unused << callback->HandshakeDone();
-        }
-      }));
+  if (mTlsHandshakeCallback) {
+    auto callback = std::move(mTlsHandshakeCallback);
+    Unused << callback->HandshakeDone();
+  }
 }
 
 void nsNSSSocketInfo::SetNegotiatedNPN(const char* value, uint32_t length) {
